### PR TITLE
Screens section refactoring

### PR DESCRIPTION
**Pull request by Alwox**

### CODE REVIEW

File: src/browser/components/SignpostImg.react.js [hunk 1]
```diff
@@ -16,13 +16,6 @@ export default class SignpostImg extends Component {
     delayStart: PropTypes.number
   }
 
-  static defaultProps = {
```
Comment (rcampos87) [inline src/browser/components/SignpostImg.react.js]: why deleting default props?

Comment (Alwox) [inline src/browser/components/SignpostImg.react.js]: It wasn't used anywhere.

File: src/browser/components/SignpostImg.react.js [hunk 2]
```diff
@@ -16,6 +16,13 @@ export default class SignpostImg extends Component {
     delayStart: PropTypes.number
   }
 
+  static defaultProps = {
```
Comment (rcampos87) [inline src/browser/components/SignpostImg.react.js]: 😄

File: src/browser/components/SignpostImg.react.js (hunk 1 above)
Comment (Alwox) [inline src/browser/components/SignpostImg.react.js]: Sorry, i was thinking it wasn't used anywhere but now i checked and see that React class is using it without help ;)

Comment (rcampos87) [inline src/browser/components/SignpostImg.react.js]: yes, its default values for the props. 😉

Comment (Alwox) [inline src/browser/components/SignpostImg.react.js]: now i will remember 😅

File: src/browser/components/SignpostImg.react.js [hunk 3]
```diff
@@ -112,7 +112,7 @@ export default class SignpostImg extends Component {
             };
             return <img className='footPrint' key={idx} src={imageSrc} style={imgStyle}/>;
           }
-          )}
+          )}{console.log(this.props.delayStart)}
```
Comment (michalklim) [inline src/browser/components/SignpostImg.react.js]: console.log 😨

File: src/browser/components/SkillItem.react.js [hunk 4]
```diff
@@ -31,15 +31,29 @@ export default class SkillItem extends Component {
     const STARS_LENGTH = 5;
     for (let i = STARS_LENGTH; i >= 1; i--) {
       starsLabels.push(
-          <input className={`star star-${i}`} id={`star${i}_${this.props.data.id}`} key={i} name="star" onChange={(e) => this.handleStarChange(i, e)} type="radio"/>,
-          <label className={`star star-${i}`} htmlFor={`star${i}_${this.props.data.id}`} key={i + STARS_LENGTH} />
+          <input className={`star star-${i}`}
+                 id={`star${i}_${this.props.data.id}`}
+                 key={i}
+                 name="star"
+                 onChange={(e) => this.handleStarChange(i, e)}
```
Comment (michalklim) [inline src/browser/components/SkillItem.react.js]: Do we have to pass `i` here?

File: src/browser/components/SkillItem.react.js [hunk 5]
```diff
@@ -31,15 +31,29 @@ export default class SkillItem extends Component {
...
     return (
       <li>
-        {this.state.hintDisplayed ? <div className="hint">{this.props.data.name}</div> : false}
-        <img alt={this.props.data.name} onMouseEnter={()=>this.showHint()} onMouseLeave={()=>this.hideHint()} src={this.props.data.src}/>
+        {this.state.hintDisplayed ?
+          <div className="hint">{this.props.data.name}</div> : false}
+
+        <img alt={this.props.data.name}
+             onMouseEnter={()=>this.showHint()}
+             onMouseLeave={()=>this.hideHint()}
```
Comment (michalklim) [inline src/browser/components/SkillItem.react.js]: Unneeded arrow function here. I think `this.showHint` and `this.hideHint` would be enough

File: src/browser/screens/Page.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+export default class Screens extends Component {
+  static propTypes = {
+    actions: PropTypes.object.isRequired,
+    candidate: PropTypes.object.isRequired,
+    config: PropTypes.object.isRequired,
+    screens: PropTypes.object.isRequired
+  }
+
+  getPageOffset() {
+    const {currentScreen} = this.props.screens;
```
Comment (michalklim) [inline src/browser/screens/Page.js]: Maybe it is unnecessary to declare currentScreen as a const here

File: src/browser/screens/backpack/BackpackScreen.js [hunk 7]
```diff
@@ -34,8 +33,9 @@ export default class Third2Screen extends React.PureComponent {
         </div>
         <BackpackList />
         <BagImg />
-        <SignpostImg {...this.props} animate={this.props.isCurrent} />
-        <Chevron isAnimated onClick={e => this.proceed(e)} />
+        <SignpostImg {...this.props} animate={this.props.screens.currentScreen === 3} />
+        <Chevron isAnimated
+                 onClick={e => this.proceed(e)} />
```
Comment (michalklim) [inline src/browser/screens/backpack/BackpackScreen.js]: unnecessary function here. `this.proceed` would be enough

File: src/browser/screens/skills/SkillsScreen.js [hunk 8]
```diff
@@ -83,22 +82,32 @@ export default class FourthScreen extends React.PureComponent {
...
               >
-                <img alt="Other skills" src="../../../assets/img/skills/skills_gears.svg"/>
+                <img alt="Other skills"
+                     src="../../../assets/img/skills/skills_gears.svg"/>
               </div>
               <div className="otherSkill-text">
-                <textarea cols="16" maxLength="500" onChange={e => this.handleKeyUp(e)} placeholder="Do You have other skills?" ref="otherSkill" rows="2" tabIndex="-1"/>
+                <textarea cols="16"
+                          maxLength="500"
+                          onChange={e => this.handleKeyUp(e)}
```
Comment (michalklim) [inline src/browser/screens/skills/SkillsScreen.js]: The unnecessary anonymous function here. `this.handleKeyUp` would do the thing

File: src/browser/screens/waysInfo/WaysInfoScreen.js [hunk 9]
```diff
@@ -28,10 +28,15 @@ export default class SecondScreen extends React.PureComponent {
             <span>There are</span>
             <span><strong>several ways</strong></span>
             <span>to the <strong>top</strong></span>
-            <button onClick={e => this.proceed(e)} tabIndex="-1">choose your path</button>
-            <Chevron isAnimated onClick={e => this.proceed(e)}/>
+
+            <button onClick={e => this.proceed(e)}
```
Comment (michalklim) [inline src/browser/screens/waysInfo/WaysInfoScreen.js]: The unnecessary anonymous function here. `this.proceed` would do the thing

File: src/browser/screens/waysInfo/WaysInfoScreen.js [hunk 10]
```diff
@@ -28,10 +28,15 @@ export default class SecondScreen extends React.PureComponent {
...
+                    tabIndex="-1">choose your path</button>
+
+            <Chevron isAnimated
+                     onClick={e => this.proceed(e)}/>
```
Comment (michalklim) [inline src/browser/screens/waysInfo/WaysInfoScreen.js]: The unnecessary anonymous function here. this.proceed would do the thing

File: src/browser/components/SignpostImg.react.js (hunk 3 above)
Comment (Alwox) [inline src/browser/components/SignpostImg.react.js]: ups 😅

File: src/browser/components/SkillItem.react.js (hunk 4 above)
Comment (Alwox) [inline src/browser/components/SkillItem.react.js]: i think yes - it distinguish stars (but it look like we don't need e)

File: src/browser/components/SkillItem.react.js (hunk 5 above)
Comment (Alwox) [inline src/browser/components/SkillItem.react.js]: > Warning: setState(...): Cannot update during an existing state transition (such as within `render` or another component's constructor). Render methods should be a pure function of props and state; constructor side-effects are an anti-pattern, but can be moved to `componentWillMount`.

Comment (Alwox) [inline src/browser/components/SkillItem.react.js]: But then we need bind this - so which its better?

File: src/browser/components/SkillItem.react.js (hunk 4 above)
Comment (michalklim) [inline src/browser/components/SkillItem.react.js]: I think we can distinguish stars from `e` alone. Then `this.handleStarChange` will do the trick

**Pull request merged**